### PR TITLE
[SDL 0173] - Read Generic Network Signal Data Clarification

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
## Introduction

* Explicit mention in MOBILE_API new not mandatory parameter oemCustomDataType. This is needed to match [proxy side implementation](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0173-Read-Generic-Network-Signal-data.md#update-vehicledataresult-to-support-custom-vehicle-data-items-by-adding)
* Extending VehicleDataType with additional value VEHICLEDATA_OEM_CUSTOM_DATA in MOBILE and HMI APIs for sending subscription/unsubscription result to mobile. This is clarification detail for following [point in the original proposal(SDL-0173)](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0173-Read-Generic-Network-Signal-data.md#impact-on-existing-code)
 - Remove @nonnull directive from oemCustomDataType parameter in Proxy library since the parameter is optional

## Motivation

The accepted proposal should  explicitly contain API changes.

#### API changes required for custom vehicle data:

- VehicleDataResult in MOBILE API will be extended with non-mandatory parameter `oemCustomDataType` 

```
   <struct name="VehicleDataResult" since="2.0">
        <description>Individual published data request result</description>
        <param name="dataType" type="VehicleDataType" mandatory="true">
            <description>Defined published data element type.</description>
        </param>
        <param name="resultCode" type="VehicleDataResultCode" mandatory="true">
            <description>Published data result code.</description>
        </param>
+        <param name="oemCustomDataType" type="String" mandatory="false" since="6.0">
+            <description>Type of requested oem specific parameter </description>
+        </param>
    </struct>
```

`VehicleDataResult` in MOBILE API will be extended with non-mandatory parameter `oemCustomDataType`

`VehicleDataType` will be extended with `VEHICLEDATA_OEM_CUSTOM_DATA` value in HMI and MOBILE APIs.
```
 <enum name="VehicleDataType" since="2.0">
        <description>Defines the data types that can be published and subscribed to.</description>
        <element name="VEHICLEDATA_GPS">
        ...
        <element name="VEHICLEDATA_CLOUDAPPVEHICLEID" since="5.1"/>
 +       <element name="VEHICLEDATA_OEM_CUSTOM_DATA" since="6.0"/>
 </enum>
```

For OEM specific custom vehicle data items, `oemCustomDataType` will contain a type of OEM specific vehicle data (from schema), and `dataType` will be `VEHICLEDATA_OEM_CUSTOM_DATA`.
For vehicle data items from RPCSpec, `oemCustomDataType` will be omitted, and `dataType` will contain appropriate data type from `VehicleDataType` enum.

### Update VehicleDataResult to support custom vehicle data items by adding:
```
-  public VehicleDataResult(@NonNull String oemCustomDataType, @NonNull VehicleDataResultCode resultCode){ 
+ public VehicleDataResult( String oemCustomDataType, VehicleDataResultCode resultCode){ 
```

```
-  public void setOEMCustomVehicleDataType(@NonNull String oemCustomDataType) { 
+  public void setOEMCustomVehicleDataType(String oemCustomDataType) { 
```

## Potential downsides
N/A Changes clarifies proposal, but not introduce changes 

## Alternatives considered

N/A Changes clarifies proposal, but not introduce changes  

## Impact on existing code

Impacts only Vehicle data subscription commands

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
@theresalech please include this Proposal clarification for SDLC review

**Comment by mrapitis** [comment]
@theresalech Ford has reviewed and approved of these changes.

File: proposals/0173-Read-Generic-Network-Signal-data.md [hunk 1]
```diff
@@ -648,6 +648,44 @@ Consider **gps** as API vehicle data Item, and **HEAD_LAMP_STATUS** as vehicle d
...
+        <element name="VEHICLEDATA_GPS">
+        ...
+        <element name="VEHICLEDATA_CLOUDAPPVEHICLEID" since="5.1"/>
+ +       <element name="VEHICLEDATA_OEM_CUSTOM_DATA" since="6.0"/>
+ </enum>
+```
+
+
+
+For oem specific custom vehicle data items, `oemCustomDataType` will contain type of OEM specific vehicle data (from schema), and `dataType` will be `VEHICLEDATA_OEM_CUSTOM_DATA`.
```
Comment (theresalech) [inline proposals/0173-Read-Generic-Network-Signal-data.md]:
```suggestion
For OEM specific custom vehicle data items, `oemCustomDataType` will contain type of OEM specific vehicle data (from schema), and `dataType` will be `VEHICLEDATA_OEM_CUSTOM_DATA`.
```

File: proposals/0173-Read-Generic-Network-Signal-data.md [hunk 2]
```diff
@@ -648,6 +648,44 @@ Consider **gps** as API vehicle data Item, and **HEAD_LAMP_STATUS** as vehicle d
...
+For vehicle data items from RPCSpec, `oemCustomDataType` will be omitted, and `dataType` will contain appropriate data type from `VehicleDataType` enum.
```
Comment (theresalech) [inline proposals/0173-Read-Generic-Network-Signal-data.md]:
```suggestion
For vehicle data items from RPC Spec, `oemCustomDataType` will be omitted, and `dataType` will contain appropriate data type from `VehicleDataType` enum.
```

File: proposals/0173-Read-Generic-Network-Signal-data.md [hunk 3]
```diff
@@ -648,6 +648,44 @@ Consider **gps** as API vehicle data Item, and **HEAD_LAMP_STATUS** as vehicle d
...
+   <struct name="VehicleDataResult" since="2.0">
+        <description>Individual published data request result</description>
+        <param name="dataType" type="VehicleDataType" mandatory="true">
+            <description>Defined published data element type.</description>
+        </param>
+        <param name="resultCode" type="VehicleDataResultCode" mandatory="true">
+            <description>Published data result code.</description>
+        </param>
++        <param name="oemCustomDataType" type="String" mandatory="false" since="6.0">
++            <description>Type of requested oem specific parameter </description>
```
Comment (theresalech) [inline proposals/0173-Read-Generic-Network-Signal-data.md]:
```suggestion
+            <description>Type of requested OEM specific parameter </description>
```

**Comment by LuxoftAKutsan** [comment]
@theresalech  PR changed according to comments

**Comment by theresalech** [comment]
@LuxoftAKutsan can you please include the `Proposed solution`, `Potential downsides`, `Impact on existing code`, and `Alternatives considered` sections?  My apologies, it appears some of those were also missed in the example I provided you.

**Comment by LuxoftAKutsan** [comment]
@theresalech  Required sections added to proposal

**Comment by LuxoftAKutsan** [comment]
@theresalech please review.

**Pull request merged**